pkg/build/docker_go.go

```diff
@@ -228,12 +211,15 @@ func (b *DockerGoBuilder) Build(ctx context.Context, in *api.BuildInput, ow *rpc
 		args["RUNTIME_IMAGE"] = &cfg.RuntimeImage
 	}
 
+	cacheimage := fmt.Sprintf("tg-gobuildcache-%s", in.TestPlan)
 	var baseimage string
+	var alreadyCached bool
```

Comment:
I am not sure I understand the purpose of this variable... it seems that in the `if condition` below it is always set to `true`.

I fail to see how `cfg.EnableGoBuildCache && !alreadyCached` can happen.

Reply:
If `EnableGoBuildCache` is set for the first time and there's no cached image, we cache it. The other command is just to remove the cache. By removing that condition, we'd be updating the cache image every time, making it a rolling cache.